src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -359,50 +376,170 @@ internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<Functi
             return triggers;
         }
 
-        private async Task<Dictionary<string, string>> ReadDurableTaskConfig()
+        private async Task<GlobalDurableConfig> ReadDurableTaskConfig()
         {
             var hostOptions = _applicationHostOptions.CurrentValue.ToHostOptions();
             string hostJsonPath = Path.Combine(hostOptions.RootScriptPath, ScriptConstants.HostMetadataFileName);
-            var config = new Dictionary<string, string>();
+
+            string durableMajorVersion;
             if (FileUtility.FileExists(hostJsonPath))
             {
-                var json = JObject.Parse(await FileUtility.ReadAsync(hostJsonPath));
+                var hostJson = JObject.Parse(await FileUtility.ReadAsync(hostJsonPath));
+
+                durableMajorVersion = await GetDurableMajorVersionAsync(hostJson, hostOptions);
 
                 // get the DurableTask extension config section
                 JToken extensionsValue;
-                if (json.TryGetValue("extensions", StringComparison.OrdinalIgnoreCase, out extensionsValue) && extensionsValue != null)
```

Comment:
This code is hard to read. Can you do something like:

```csharp
private async Task<GlobalDurableConfig> ReadDurableTaskConfig()
{
	JObject hostJson = null;
	JObject durableHostConfig = null;
	var hostOptions = _applicationHostOptions.CurrentValue.ToHostOptions();
	string hostJsonPath = Path.Combine(hostOptions.RootScriptPath, ScriptConstants.HostMetadataFileName);
	if (FileUtility.FileExists(hostJsonPath))
	{
		hostJson = JObject.Parse(await FileUtility.ReadAsync(hostJsonPath));
		
		// get the DurableTask extension config section
		if (hostJson != null &&
			hostJson.TryGetValue("extensions", StringComparison.OrdinalIgnoreCase, out JObject extensionsValue))
		{
			// we will allow case insensitivity given it is likely user hand edited
			// see https://github.com/Azure/azure-functions-durable-extension/issues/111
			var extensions = extensionsValue as JObject;
			if (extensions != null &&
				extensions.TryGetValue(DurableTask, StringComparison.OrdinalIgnoreCase, out JObject durableTaskValue))
			{
				durableHostConfig = durableTaskValue as JObject;
			}
		}
	}
	
	var durableMajorVersion = await GetDurableMajorVersionAsync(hostJson, hostOptions);
	if (durableMajorVersion == null || durableMajorVersion.Equals("1"))
	{
		return GetDurableV1Config(durableHostConfig);
	}
	else
	{
		// v2 or greater
		return GetDurableV2Config(durableHostConfig);
	}
}
```

Reply:
Took most of your suggestion.